.circleci/config.yml

```diff
@@ -84,7 +86,361 @@ commands:
+        type: integer
+    parallelism: << parameters.parallelism >>
+    docker:
+      - image: cimg/node:<< pipeline.parameters.node-version >>
+    steps:
+      - checkout
+      - yarn-install
+
+      - run:
+          name: "Compile TS"
+          command: yarn workspaces foreach --topological-dev --recursive --verbose --from "<< parameters.workspace >>" run build:ts
+
+      - run:
+          name: "Run tests"
+          command: |
+            export PATH=$PATH:$PWD/node_modules/.bin
+            _dir=$(yarn workspace "<< parameters.workspace >>" exec pwd)
+            echo "Workspace: $_dir"
+            cd $_dir
+            _tests=$(circleci tests glob 'test/**/*.test.ts' | circleci tests split --split-by=timings)
+            echo "Tests:"
+            for _test in $_tests; do
+              echo $_test
+            done
+            mocha --timeout 10000 --require ts-node/register --reporter mocha-junit-reporter --reporter-options mochaFile=/tmp/junit/report.xml --exit $_tests
+
+      - store_test_results:
+          path: "/tmp/junit"
+
+  test-core-modules:
+    parameters:
+      workspace:
+        type: string
+      parallelism:
+        type: integer
+    parallelism: << parameters.parallelism >>
+    docker:
+      - image: cimg/node:<< pipeline.parameters.node-version >>
+    steps:
+      - checkout
+      - install-foundry
+      - yarn-install
+
+      - restore_cache:
+          keys:
+            - hardhat-{{ .Environment.SOLC_VERSION }}
+
+      - run:
+          name: "Compile TS"
+          command: yarn workspaces foreach --topological-dev --recursive --verbose --from "<< parameters.workspace >>" run build:ts
+
+      - run:
+          name: "Run tests"
+          environment:
+            REPORT_GAS: true
+          command: |
+            export PATH=$PATH:$PWD/node_modules/.bin
+            _dir=$(yarn workspace "<< parameters.workspace >>" exec pwd)
+            echo "Workspace: $_dir"
+            cd $_dir
+            _tests=$(circleci tests glob 'test/**/*.test.ts' | circleci tests split --split-by=timings)
+            echo "Tests:"
+            for _test in $_tests; do
+              echo $_test
+            done
+            mocha --timeout 10000 --require hardhat/register --reporter mocha-junit-reporter --reporter-options mochaFile=/tmp/junit/report.xml --exit $_tests
+
+      - store_test_results:
+          path: "/tmp/junit"
+
+  test-core-contracts:
+    parameters:
+      workspace:
+        type: string
+      parallelism:
+        type: integer
+    parallelism: << parameters.parallelism >>
+    docker:
+      - image: cimg/node:<< pipeline.parameters.node-version >>
+    steps:
+      - checkout
+      - yarn-install
+
+      - restore_cache:
+          keys:
+            - hardhat-{{ .Environment.SOLC_VERSION }}
+
+      - run:
+          name: "Compile TS"
+          command: yarn workspaces foreach --topological-dev --recursive --verbose --from "<< parameters.workspace >>" run build:ts
```

Comment:
Technically core contracts don't depend on anything so there is no need to go topological here